xmodule/modulestore/__init__.py

```diff
@@ -313,13 +319,54 @@ def send_pre_publish_signal(self, bulk_ops_record, course_id):
         if signal_handler and bulk_ops_record.has_publish_item:
             signal_handler.send("pre_publish", course_key=course_id)
 
+    def create_catalog_data_for_signal(self, course_id):
+        """
+        Creates a CourseCatalogData object from a course id by fetching from the module store
+
+        Arguments:
+            course_id (CourseKey): id of the course to process
+
+        Returns:
+            CourseCatalogData
+        """
+        # Most of this is copied from CourseOverview. We cannot import it
+        # because it would lead to circular imports
+        course_from_store = self.get_course(course_id)
+        number = number_for_course_location(course_from_store.location)
+        catalog_visibility = course_from_store.catalog_visibility
+
+        # see CourseDetail.fetch_about_attribute
+        effort_usage_key = course_id.make_usage_key('about', 'effort')
+        try:
+            effort = self.get_item(effort_usage_key).data
+        except ItemNotFoundError:
+            effort = None
+        return CourseCatalogData(
+            course_key=course_id.for_branch(None),
```

Comment:
Does this need a comment as well, like the one in `send_bulk_published_signal`? And is it always true that this method will be called during a publish? (Maybe there needs to be a guard, or a different method name.)